tcpping/tcpping.go

```diff
@@ -0,0 +1,95 @@
+package tcpping
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	errors "github.com/pkg/errors"
+	tcp "github.com/tevino/tcp-shaker"
+)
+
+type Result struct {
+	Address string
+	Latency int
+	PingErr error
+}
+
+func (cr *Result) String() string {
+	return fmt.Sprintf(
+		"Result{address: %s, latency: %d, error: %s}\n",
+		cr.Address, cr.Latency, cr.PingErr)
+}
+
+func runCheck(c *tcp.Checker, address string, resChan chan Result, timeout time.Duration) {
+	// mesaure RTT
+	start := time.Now()
+	// TCP Ping
+	err := c.CheckAddr(address, timeout)
```

Comment:
Handle error here immediately.

Reply:
But I don't want to return a `Result` struct with the error, so it can be returned. Stopping here makes no sense because the RPC call response wont include info about why it failed.